File: WFInfo/Sheets.vb
Imports Google.Apis.Auth.OAuth2
Imports Google.Apis.Services
Imports Google.Apis.Sheets.v4
Imports Google.Apis.Sheets.v4.Data

Class Sheets

    Private cred As GoogleCredential
    Private ReadOnly sheetID = "1uAbqfwBYrcqlWCJad4juankmu9-_UqzBmd7XtrrrwkM"
    Private service As SheetsService

    Public Sub New()
        cred = GoogleCredential.FromJson(My.Resources.google_creds).CreateScoped({SheetsService.Scope.SpreadsheetsReadonly})
        service = New SheetsService(New BaseClientService.Initializer() With
        {
            .HttpClientInitializer = cred,
            .ApplicationName = "WFInfo"
        })

    End Sub

    Public Function GetSheet(range As String) As IList(Of IList(Of Object))
        Dim response As ValueRange = service.Spreadsheets.Values.Get(sheetID, range).Execute()
        Return response.Values
    End Function
End Class

File: WFInfo/sheets.cs
using Google.Apis.Auth.OAuth2;
using Google.Apis.Services;
using Google.Apis.Sheets.v4;
using Google.Apis.Sheets.v4.Data;
using System.Collections.Generic;

namespace WFInfo
{
    internal class Sheets
    {
        private readonly GoogleCredential Cred;
        private readonly string SheetID = "1uAbqfwBYrcqlWCJad4juankmu9-_UqzBmd7XtrrrwkM";
        private readonly SheetsService service;

        public Sheets()
        {
            Cred = GoogleCredential.FromJson(Properties.Resources.google_creds).CreateScoped(SheetsService.Scope.SpreadsheetsReadonly);
            service = new SheetsService(new BaseClientService.Initializer()
            {
                HttpClientInitializer = Cred,
                ApplicationName = "WFInfo"
            });
        }


        public IList<IList<object>> GetSheet(string range)
        {
            ValueRange response;
            try
            {
                response = service.Spreadsheets.Values.Get(SheetID, range).Execute();
            }
            catch (System.Exception)
            {
                throw;
            }
            return response.Values;

        }
    }
}